src/System/Random/Internal.hs

```diff
@@ -667,48 +678,89 @@ class UniformRange a where
   -- > isInRange (lo, hi) lo' && isInRange (lo, hi) hi' && isInRange (lo', hi') x
   -- > ==> isInRange (lo, hi) x
   --
+  -- There is a default implementation of 'isInRange' via 'Generic'.
+  --
   -- @since 1.3.0
   isInRange :: (a, a) -> a -> Bool
 
-  default isInRange :: Ord a => (a, a) -> a -> Bool
-  isInRange (a, b) x = min a b <= x && x <= max a b
+  default uniformRM :: (StatefulGen g m, Generic a, GUniformRange (Rep a)) => (a, a) -> g -> m a
+  uniformRM (a, b) = fmap to . (`runContT` pure) . guniformRM (from a, from b)
+  {-# INLINE uniformRM #-}
+
+  default isInRange :: (Generic a, GUniformRange (Rep a)) => (a, a) -> a -> Bool
+  isInRange (a, b) x = gisInRange (from a, from b) (from x)
   {-# INLINE isInRange #-}
 
+class GUniformRange f where
+  guniformRM :: StatefulGen g m => (f a, f a) -> g -> ContT r m (f a)
+  gisInRange :: (f a, f a) -> f a -> Bool
+
+instance GUniformRange f => GUniformRange (M1 i c f) where
+  guniformRM (M1 a, M1 b) = fmap M1 . guniformRM (a, b)
+  {-# INLINE guniformRM #-}
+  gisInRange (M1 a, M1 b) (M1 x) = gisInRange (a, b) x
+
+instance UniformRange a => GUniformRange (K1 i a) where
+  guniformRM (K1 a, K1 b) = fmap K1 . lift . uniformRM (a, b)
+  {-# INLINE guniformRM #-}
+  gisInRange (K1 a, K1 b) (K1 x) = isInRange (a, b) x
+
+instance GUniformRange U1 where
+  guniformRM = const $ const $ return U1
+  {-# INLINE guniformRM #-}
+  gisInRange = const $ const True
+
+instance (GUniformRange f, GUniformRange g) => GUniformRange (f :*: g) where
+  guniformRM (x1 :*: y1, x2 :*: y2) g =
+    (:*:) <$> guniformRM (x1, x2) g <*> guniformRM (y1, y2) g
+  {-# INLINE guniformRM #-}
+  gisInRange (x1 :*: y1, x2 :*: y2) (x3 :*: y3) =
+    gisInRange (x1, x2) x3 && gisInRange (y1, y2) y3
+
+isInRangeOrd :: Ord a => (a, a) -> a -> Bool
+isInRangeOrd (a, b) x = min a b <= x && x <= max a b
```

Comment:
I still worry about this even though it's not a change since we had `isInRange` with the same definition so the boat has sailed.  BTW what do we get with `isInRange (127 :: Int8, -128 :: Int8)` - it seems

```
> fst $ (uniformR (127 :: Int8, -128)) (mkStdGen 42)
fst $ (uniformR (127 :: Int8, -128)) (mkStdGen 42)
-80
```

I would have expected 127 or -128 as I think of `Int8` as a clock - perhaps I shouldn't. Anyhow nothing to do with this change.